ONSdigital/ras-rm-performance-tests
# Extend to harvest url and 302 gets

# What and why?
This PR adds a havest process to allow dynamic hyperlinks to be used. ( in this scenario case_id, would not be the same, so we can't hard coded a url into the request file). It also introduces a 302 get assert, so we don't have to follow redirects. These 2 pieces combined allow eq redirects in frontstage to be tested

# How to test?
The easiest way to test it is through concourse, I have even left the performance branch pointing at latest (N.B someone might change it back if this review hangs around)(https://concourse.rmrasbs.gcp.onsdigital.uk/teams/development/pipelines/performance/jobs/deploy-locust/builds/29#L668787b2:2) you can find a successful run with the update from this PR in the folder 24-07-08-06-21 on performance

You can test it locally if you want, but the repo has become cumbersome to do so (well before this PR). This PR from Mark is useful https://github.com/ONSdigital/ras-rm-performance-tests/pull/26 (i.e the exports and the pipfile, bs4 needs to be added to it) 
When it comes to vars they are a bit all over the place when used locally

```requests_file = '/mnt/locust/' + os.getenv('requests_file')``` should just be ```requests_file ='requests.json'```

anywhere /mnt/locust/ is used, just delete it 

```config = json.load(open("/mnt/locust/collection-exercise-config.json")) ```

becomes 

```config = json.load(open("collection-exercise-config.json"))```

and in the requests.json file the assert text for eq can be shortened to ```session?token=```


# Trello
